### PR TITLE
Add SNO upgrade workflows

### PR DESCRIPTION
**Pull request by vrutkovs**
This setups SNO cluster on AWS and Azure from `release:initial` and upgrades it
to `release:latest`. Disruption tests are skipped, as they are not
expected to pass in SNO configuration.

Split out from #16290 to speed up review.

TODO:
* [x] [Skip disruption tests](https://github.com/openshift/origin/pull/26061)
* [x] Fix `storage` operator on AWS - https://bugzilla.redhat.com/show_bug.cgi?id=1947402

### CODE REVIEW

**Comment by vrutkovs** [comment]
Reference: https://issues.redhat.com/browse/OTA-379

File: ci-operator/step-registry/openshift/upgrade/aws/single-node/openshift-upgrade-aws-single-node-workflow.yaml
Comment (wking) [inline ci-operator/step-registry/openshift/upgrade/aws/single-node/openshift-upgrade-aws-single-node-workflow.yaml:32]: I realize that `TEST_SKIPS` was my idea from #12233, but having lived with it in the proxy space for a while now, an important limitation is that it's a pain to clobber this per-job for tests which are fixed in one version of OpenShift but not in older versions.  So while I have no problem with this sort of thing in the short term, there should be a bug or bugs about adding skip logic to openshift/origin to avoid things we know will never work when there is only a single control-plane node (the Kube/OpenShift(/OAuth?) API stuff) or only one node period (the ingress/LB/registry stuff).  Ideally we create that bug (or bugs) before landing this PR, and link them from comments in the YAML.

File: ci-operator/step-registry/single-node/conf/single-node-conf-commands.sh
Comment (wking) [inline ci-operator/step-registry/single-node/conf/single-node-conf-commands.sh:36]: nit: "AWS and Azure", not GCP.  Probably safer to use something like:

```python
raise ValueError("Single node configuration does not have a configured instance type for {}".format(platform))
```

File: ci-operator/step-registry/openshift/upgrade/aws/single-node/openshift-upgrade-aws-single-node-workflow.yaml
Comment (vrutkovs) [reply ci-operator/step-registry/openshift/upgrade/aws/single-node/openshift-upgrade-aws-single-node-workflow.yaml:32]: >there should be a bug or bugs about adding skip logic to openshift/origin

Good idea - filed https://bugzilla.redhat.com/show_bug.cgi?id=1947809

**Comment by vrutkovs** [comment]
/cc @romfreiman @omertuc @smarterclayton @droslean 

Tests still fail as a) AWS/GCP needs storage operator fixed, b) origin tests should skip disruption tests (I'll create a new origin PR to skip those)

File: ci-operator/step-registry/single-node/conf/azure/single-node-conf-azure-commands.sh
Comment (omertuc) [inline ci-operator/step-registry/single-node/conf/azure/single-node-conf-azure-commands.sh:34]: nit: Just noticed this error is very misleading right now. Please fix it to `This step only applies to Azure, please use the correct step for your platform`.

Also please modify the message in the corresponding {aws,gcp} steps (if you don't want to do it in this PR, let me know, I'll do it separately)

File: ci-operator/step-registry/openshift/upgrade/azure/single-node/openshift-upgrade-azure-single-node-workflow.yaml
Comment (omertuc) [inline ci-operator/step-registry/openshift/upgrade/azure/single-node/openshift-upgrade-azure-single-node-workflow.yaml:25]: `/s/AWS/Azure`

File: ci-operator/step-registry/single-node/conf/azure/single-node-conf-azure-commands.sh
Comment (vrutkovs) [reply ci-operator/step-registry/single-node/conf/azure/single-node-conf-azure-commands.sh]: Done. I'd rather not touch other steps in this commit to make it revertable.

GCP/AWS text can be updated later, its not critical

File: ci-operator/step-registry/single-node/conf/azure/single-node-conf-azure-commands.sh
Comment (omertuc) [inline ci-operator/step-registry/single-node/conf/azure/single-node-conf-azure-commands.sh:34]: nit: sorry, about this again - you forgot to remove the assertion, so this branch will never happen

File: ci-operator/step-registry/single-node/conf/azure/single-node-conf-azure-commands.sh
Comment (omertuc) [reply ci-operator/step-registry/single-node/conf/azure/single-node-conf-azure-commands.sh]: Sure, created https://github.com/openshift/release/pull/17694

File: ci-operator/step-registry/single-node/conf/azure/single-node-conf-azure-commands.sh
Comment (vrutkovs) [reply ci-operator/step-registry/single-node/conf/azure/single-node-conf-azure-commands.sh:33]: Updated (removed assertion to get a better error message)

**Comment by vrutkovs** [comment]
/hold

I'll create a new PR to fix `openshift-e2e-test` typo

File: ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml
Comment (romfreiman) [inline ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml:5905]: we called in single-node everywhere (rather than sno)

File: ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh
Comment (romfreiman) [inline ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh:127]: is this in purpose?

File: ci-operator/step-registry/openshift/upgrade/aws/single-node/openshift-upgrade-aws-single-node-workflow.yaml
Comment (romfreiman) [inline ci-operator/step-registry/openshift/upgrade/aws/single-node/openshift-upgrade-aws-single-node-workflow.yaml:26]: is loki is something specific here? Just so i understand, or this is the standard flow

**Comment by romfreiman** [comment]
Lets add those jobs here as well: https://testgrid.k8s.io/redhat-single-node

File: ci-operator/step-registry/openshift/upgrade/aws/single-node/openshift-upgrade-aws-single-node-workflow.yaml
Comment (omertuc) [reply ci-operator/step-registry/openshift/upgrade/aws/single-node/openshift-upgrade-aws-single-node-workflow.yaml:26]: Copied from regular upgrade workflow

File: ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh
Comment (vrutkovs) [reply ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh:127]: Fixes a typo in previous PR - https://github.com/openshift/release/pull/17823

File: ci-operator/step-registry/openshift/upgrade/aws/single-node/openshift-upgrade-aws-single-node-workflow.yaml
Comment (vrutkovs) [reply ci-operator/step-registry/openshift/upgrade/aws/single-node/openshift-upgrade-aws-single-node-workflow.yaml:26]: This step would setup promtail on the cluster, sending container and journald logs to the central repo, which can be queried by Grafana.

This is a standard practice for upgrade workflows, as intermediate containers can be lost during reboot (or the node may never come back)

File: ci-operator/step-registry/single-node/conf/azure/single-node-conf-azure-commands.sh
Comment (wking) [inline ci-operator/step-registry/single-node/conf/azure/single-node-conf-azure-commands.sh:7]: I expect there to be one set of folks maintaining these single-node conf step(s), so it might we worth a single `single-node/conf` command instead of the platform-specific sub-steps.  On the other hand, I'm not in the maintainer set, so doesn't matter for me ;).

**Review by wking — APPROVED** [review]
Looks pretty good to me.  Remaining wrinkles can be ironed out in follow-up work.

/lgtm
/retest

**Comment by vrutkovs** [comment]
/hold cancel

All issues were resolved

**Pull request merged**
